LoopKitUI/View Controllers/ChartsTableViewController.swift

```diff
@@ -0,0 +1,151 @@
+//
+//  ChartsTableViewController.swift
+//  LoopKitUI
+//
+//  Copyright © 2017 LoopKit Authors. All rights reserved.
+//
+
+import UIKit
+import HealthKit
+
+/// Abstract class providing boilerplate setup for chart-based table view controllers
+open class ChartsTableViewController: UITableViewController, UIGestureRecognizerDelegate {
+
+    public var preferredGlucoseUnit: HKUnit?
+
+    open override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        if let unit = preferredGlucoseUnit {
+            self.charts.setGlucoseUnit(unit)
+        }
+
+        let gestureRecognizer = UILongPressGestureRecognizer()
+        gestureRecognizer.delegate = self
+        gestureRecognizer.minimumPressDuration = 0.1
+        gestureRecognizer.addTarget(self, action: #selector(handlePan(_:)))
+        charts.gestureRecognizer = gestureRecognizer
+    }
+
+    open override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+
+        if !visible {
+            charts.didReceiveMemoryWarning()
+        }
+    }
+
+    open override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        visible = true
+    }
+
+    open override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        visible = false
+    }
+
+    open override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+
+        reloadData(animated: false)
+    }
+
+    open override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        charts.traitCollection = traitCollection
+    }
+
+    // MARK: - State
+    // This function should only be called from the main thread
+    public func unitPreferencesDidChange(to unit: HKUnit?) {
+        if let unit = unit {
+            self.charts.setGlucoseUnit(unit)
+            self.glucoseUnitDidChange()
+        }
+        self.reloadData()
+    }
```

Comment:
This is a slightly modified version of what was previously present (essentially, it's now a public function that is called by the parent instead of an `@objc` function that's called by an observer)